File: src/main/java/pl/patrykdolata/lifemanager/domain/UserEntity.java
package pl.patrykdolata.lifemanager.domain;

import lombok.Data;

import javax.persistence.*;
import javax.validation.constraints.Email;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

@Data
@Entity
@Table(name = "user")
public class UserEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull
    @Size(min = 3, max = 50)
    @Column(name = "username", nullable = false, length = 50, unique = true)
    private String username;

    @NotNull
    @Size(min = 60, max = 60)
    @Column(name = "password", nullable = false, length = 60)
    private String password;

    @NotNull
    @Size(min = 3, max = 50)
    @Column(name = "first_name", nullable = false, length = 50)
    private String firstName;

    @NotNull
    @Size(min = 3, max = 50)
    @Column(name = "last_name", nullable = false, length = 50)
    private String lastName;

    @Email
    @NotNull
    @Size(min = 5, max = 254)
    @Column(name = "email", nullable = false, length = 254, unique = true)
    private String email;

    @NotNull
    @Column(name = "activated", nullable = false)
    private Boolean activated;

    @Size(max = 30)
    @Column(name = "activation_key", length = 30)
    private String activationKey;

    @Size(max = 30)
    @Column(name = "reset_key", length = 30)
    private String resetKey;

    @Column(name = "created_at")
    private Long createdAt;
}

File: src/main/kotlin/pl/patrykdolata/lifemanager/domain/UserEntity.kt
package pl.patrykdolata.lifemanager.domain

import pl.patrykdolata.lifemanager.domain.UserEntity.Companion.TABLE_NAME
import javax.persistence.*
import javax.validation.constraints.Email
import javax.validation.constraints.Size

@Entity
@Table(name = TABLE_NAME)
class UserEntity : AbstractEntity<Long>() {

    companion object {
        const val TABLE_NAME = "user"
    }

    @Size(min = 3, max = 50)
    @Column(name = "username", length = 50, unique = true)
    var username: String = ""

    @Size(min = 60, max = 60)
    @Column(name = "password", length = 60)
    var password: String = ""

    @Size(min = 3, max = 50)
    @Column(name = "first_name", length = 50)
    var firstName: String = ""

    @Size(min = 3, max = 50)
    @Column(name = "last_name", length = 50)
    var lastName: String = ""

    @Email
    @Size(min = 5, max = 254)
    @Column(name = "email", length = 254, unique = true)
    var email: String = ""

    @Column(name = "activated")
    var activated: Boolean = false

    @Size(max = 30)
    @Column(name = "activation_key", length = 30)
    var activationKey: String? = null

    @Size(max = 30)
    @Column(name = "reset_key", length = 30)
    var resetKey: String? = null

    @Column(name = "created_at")
    var createdAt: Long = 0

    @ManyToMany
    @JoinTable(
            name = "user_authority",
            joinColumns = [JoinColumn(name = "user_id", referencedColumnName = "id")],
            inverseJoinColumns = [JoinColumn(name = "authority", referencedColumnName = "name")]
    )
    var authorities: Set<AuthorityEntity> = emptySet()
}
